### PR TITLE
Floor bulk metadata

### PR DESCRIPTION
**Pull request by sonaisrayel**

### CODE REVIEW

File: core/bulk/floor.go [hunk 1]
```diff
@@ -72,46 +79,99 @@ func ProcessChunksFloor(c *fiber.Ctx, chunks [][]FloorUpdateRequest) error {
...
+			Device:        data.Device,
+			Floor:         data.Floor,
+			Country:       data.Country,
+			Os:            null.StringFrom(data.OS),
+			PlacementType: null.StringFrom(data.PlacementType),
+			Browser:       null.StringFrom(data.Browser),
+			RuleID:        data.RuleId,
 		})
+
+		metadata, _ := SendFloorToRT(context.Background(), data)
```
Comment (finkiel) [inline core/bulk/floor.go:114]: why the method "sendFloorToRT" is inside the method prepareDataFloor - that name implies that you are preparing the data not sending it to metaData

File: core/bulk/floor.go [hunk 2]
```diff
@@ -56,7 +63,7 @@ func ProcessChunksFloor(c *fiber.Ctx, chunks [][]FloorUpdateRequest) error {
 			return err
 		}
 
-		if err := BulkInsertMetaDataQueue(c, tx, metaDataQueue); err != nil {
+		if err := InsertRegMetaDataQueue(c, tx, metaDataQueue); err != nil {
```
Comment (finkiel) [inline core/bulk/floor.go:66]: you are sending the data again to metadata_queue table?
didn't you do it before? (inside the prepareDataFloor method

File: core/bulk/utils.go [hunk 3]
```diff
@@ -40,6 +40,58 @@ func InsertInBulk(c *fiber.Ctx, tx *sql.Tx, tableName string, columns []string,
 
 	return nil
 }
+
+func generatePlaceholders(rowCount, colCount int) string {
+	var builder strings.Builder
+
+	for i := 0; i < rowCount; i++ {
```
Comment (finkiel) [inline core/bulk/utils.go:47]: if it's possible try to avoid "for" inside a "for"
maybe by a smarter query

File: core/bulk/floor.go (hunk 1 above)
Comment (sonaisrayel) [reply core/bulk/floor.go:114]: In sendFloorToRT method I am preparing data not sending , I am changing this method name to PrepareFloorToRT

File: core/bulk/floor.go (hunk 2 above)
Comment (sonaisrayel) [reply core/bulk/floor.go:66]: no in prepareDataFloor nad preparing data , and inserting  in InsertRegMetaDataQueue method

File: core/bulk/utils.go (hunk 3 above)
Comment (sonaisrayel) [reply core/bulk/utils.go:47]: I changed it

**Review by finkiel — APPROVED** [review]

**Review by siminkovladyslav — APPROVED** [review]

**Pull request closed without merging**